src/recordpage.cpp

```diff
@@ -207,24 +207,26 @@ RecordPage::RecordPage(QWidget* parent)
     ui->outputFile->setMode(KFile::File | KFile::LocalOnly);
     ui->eventTypeBox->lineEdit()->setPlaceholderText(tr("perf defaults (usually cycles:Pu)"));
 
-    ui->launchConfigComboBox->setEditable(true);
-    ui->launchConfigComboBox->setDisabled(true);
-    ui->launchConfigComboBox->setInsertPolicy(QComboBox::InsertAtCurrent);
-    connect(ui->launchConfigComboBox->lineEdit(), &QLineEdit::editingFinished, this,
-            &RecordPage::onApplicationConfigRenamed);
-    connect(ui->launchConfigComboBox, qOverload<int>(&QComboBox::currentIndexChanged), this,
-            &RecordPage::onApplicationConfigChanged);
-    connect(ui->addConfig, &QPushButton::clicked, this, &RecordPage::onApplicationConfigAdded);
-    connect(ui->removeConfig, &QPushButton::clicked, this, &RecordPage::onApplicationConfigRemoved);
-
-    const auto updateLaunchConfig = [this] {
-        if (ui->launchConfigComboBox->count() > 0) {
-            saveApplicationConfig(ui->launchConfigComboBox->currentText());
-        }
+    auto saveFunction = [this](KConfigGroup group) {
+        group.writeEntry("params", ui->applicationParametersBox->text());
+        group.writeEntry("workingDir", ui->workingDirectory->text());
+    };
+
+    auto restoreFunction = [this](const KConfigGroup& group) {
+        ui->applicationParametersBox->setText(group.readEntry("params", ""));
+        ui->workingDirectory->setText(group.readEntry("workingDir", ""));
+        setError({});
     };
 
-    connect(ui->applicationParametersBox, &QLineEdit::editingFinished, this, updateLaunchConfig);
-    connect(ui->workingDirectory, &KUrlRequester::textChanged, this, updateLaunchConfig);
+    m_multiConfig = new MultiConfigWidget(saveFunction, restoreFunction, this);
```

Comment:
`std::move(saveFunction), std::move(restoreFunction)`

side note: does your code even compile?! you are taking an rvalue ref, but you are not moving here, so this shouldn't compile at all as-is?

Reply:
yes

Reply:
seems the compiler can detect this

Reply:
still, please move it here explicitly if you stick to `std::function`

but, quite frankly, giving this a second thought I would believe using signals instead would be much nicer:

in the config widget, you'd just emit a signal instead of calling the save/restore function

and here after constructing the widget, you'd just connect to the signals to the lambdas you have above